chat/chat-api/api/src/java/org/sakaiproject/chat2/model/TransferableChatMessage.java

```diff
@@ -0,0 +1,95 @@
+package org.sakaiproject.chat2.model;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Date;
+
+import lombok.Data;
+
+/**
+ * 
+ * Class used to transfer data through jGroups
+ *
+ */
+@Data
+public class TransferableChatMessage implements Serializable {
+
+	public enum MessageType {
+	    CHAT,
+	    HEARTBEAT,
+	    CLEAR,
+	    REMOVE;
+	}
+	
+	public MessageType type;
+	public String id;
+	public String owner;
+	public String siteId;
+	public String channelId;
+	public String content;
+	public long timestamp;
+
+	public TransferableChatMessage(ChatMessage msg) {
+		this(MessageType.CHAT, msg.getId(), msg.getOwner(), msg.getChatChannel().getContext(), msg.getChatChannel().getId(), msg.getBody(), msg.getMessageDate().getTime());
+	}
+	
+	public TransferableChatMessage(MessageType type, String id) {
+		this(type, id, null, null, null, null);
+	}
+	
+	public TransferableChatMessage(MessageType type, String id, String channelId) {
+		this(type, id, null, null, channelId, null);
+	}
+
+	public TransferableChatMessage(MessageType type, String id, String owner, String siteId, String channelId, String content) {
+		this(type, id, owner, siteId, channelId, content, (new Date()).getTime());
+	}
+	public TransferableChatMessage(MessageType type, String id, String owner, String siteId, String channelId, String content, long timestamp) {
```

Comment:
my comment was about that the timestamp param is passed into the constructor but it is never used since you always construct a new Date object.